src/unstructured_extractor.py

```
@@ -626,202 +614,475 @@ def filter_rotated_elements(
+    text = text.strip()
+    if re.match(r'^\d+\.', text): return "decimal_dot"  # 1.
+    if re.match(r'^[a-z]\)', text): return "alpha_paren"  # a)
+    if re.match(r'^\([a-z]\)', text): return "paren_alpha_paren"  # (a)
+    if re.match(r'^\(\d+\)', text): return "paren_decimal_paren"  # (1)
+    if re.match(r'^[A-Z]\.', text): return "upper_alpha_dot"  # A.
+    return "unknown"
 
-    Returns:
-        List of dicts with hierarchy metadata for each element
+
+def build_robust_hierarchy(elements: List[Element], config: ExtractionConfig) -> List[Dict]:
```

Ruff (0.14.8)

Unused function argument: `config`

(ARG001)